packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -29,102 +29,94 @@ export const styles = theme => ({
   },
 });
 
-class SpeedDialAction extends React.Component {
-  constructor(props) {
-    super();
-    this.state = {
-      tooltipOpen: props.tooltipOpen,
-    };
-  }
+const SpeedDialAction = ({
+  ButtonProps,
+  classes,
+  className,
+  delay,
+  icon,
+  id,
+  innerRef,
+  onClick,
+  onKeyDown,
+  open,
+  tooltipTitle,
+  TooltipClasses,
+  tooltipPlacement,
+  tooltipOpen: tooltipOpenProp,
+  ...other
+}) => {
+  const [tooltipOpen, setTooltipOpen] = React.useState(tooltipOpenProp);
+  const timeout = React.useRef();
+  const [prevPropOpen, setPreviousOpen] = React.useState(null);
 
-  static getDerivedStateFromProps = (props, state) => {
-    if (!props.open && state.tooltipOpen) {
-      return { tooltipOpen: false };
+  // getDerivedStateFromProps alternate
+  React.useEffect(() => {
```

Comment:
This is slightly different to getDerivedStateFromProps due to it being run after the DOM has been updated. This is the recommended alternative https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops

Reply:
I have made the changes for `getDerivedStateFromProps` alternative, can you kindly check if my approach is right. 😄